icelake/src/io/functional_writer/dispatcher_writer.rs

```diff
@@ -0,0 +1,80 @@
+use crate::io::{IcebergWriteResult, IcebergWriter, IcebergWriterBuilder};
+use crate::Result;
+use arrow_array::RecordBatch;
+use arrow_schema::SchemaRef;
+
+/// DispatcherWriter used to dispatch the partitioned and unpartitioned writer so that user can store them as a single writer.
+#[derive(Clone)]
+pub struct DispatcherWriterBuilder<UP: IcebergWriterBuilder, P: IcebergWriterBuilder> {
```

Comment:
Do we really need this? How about make partitioner a trait, so that no partition is just a special case of `Partitioner`?

Reply:
This is just a helper wrapper. Do not provide it is ok.

> How about make partitioner a trait?

If we use partitioner trait, to add partition function, I guess we need to implement it in all writer they need, such as data file writer, delta writer. But if we treat it as a special functional iceberg writer, we can combine them dynamically.

Reply:
I see. We can keep it for now since it's just a utility data structure.